python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/cli_invocation.py

```diff
@@ -55,15 +55,16 @@ def run(
     def wait(
         self, timeout: Optional[float] = None
     ) -> Iterator[Union[AssetCheckEvaluation, AssetCheckResult, AssetMaterialization, Output]]:
-        self.run_handler.wait_for_success(timeout=timeout)
-        if "run_results.json" not in self.run_handler.list_run_artifacts():
-            return
-        run_results = DbtCloudJobRunResults.from_run_results_json(
-            run_results_json=self.run_handler.get_run_results()
-        )
-        yield from run_results.to_default_asset_events(
-            client=self.client,
-            manifest=self.manifest,
-            dagster_dbt_translator=self.dagster_dbt_translator,
-            context=self.context,
-        )
+        run = self.run_handler.client.poll_run(run_id=self.run_handler.run_id, poll_timeout=timeout)
```

Comment:
It would be great if the run handler could handle polling and raising for status. Maybe we could use a context manager here?